spring-integration-core/src/test/java/org/springframework/integration/handler/LoggingHandlerTests.java

```diff
@@ -126,6 +129,23 @@ public void testChangeLevel() {
 		verify(log, times(1)).warn(Mockito.anyString());
 	}
 
+	@Test(expected = MessageHandlingException.class)
```

Comment:
If you follow with my change to the `LoggingHandler`, these tests won't path anymore.
They would need to be changed to a new logic. Although that will cover the logic you have came to us originally.

OTOH we chase a bit different pattern to verify exceptions.
See, both your tests expect the same outcome. I can change a lot of in code and in tests, too, but they will still pass.

To prevent that we do this:
```
try {
	loggingHandler.handleMessage(message);
	fail("Expected MessageHandlingException");
}
catch (Exception e) {
    assertThat(e, instanceOf(MessageHandlingException.class));
    assertThat(e.getCause(), instanceOf(IllegalStateException.class));
    assertThat(e.getMessage(), containsString("'expression' must not be null"));
}
```